addr.go

```diff
@@ -18,14 +19,16 @@ const (
 )
 
 // chooseAddr selects an Addr from the interface based on the specified Addr type.
-func chooseAddr(addrs []net.Addr, zone string, addr Addr) (netip.Addr, error) {
+func chooseAddr(addrs []net.Addr, zone string, zoneIndex int, addr Addr) (netip.Addr, error) {
```

Comment:
I feel like the code is a bit more complicated than it needs to be. Could we do the following:
- assume that `addr` does not include a zone information (or ignore it?) as when `Listen` is called, we are already working with a specific `net.Interface`
- remove `zone` and only keep `zoneIndex`
- when returning the final address (return value of type `netip.Addr`), always set the zone to `zoneIndex`

The current code feels like it's a bit too complex and was tweaked for a specific use case

Reply:
The motivation for this change is to adjust the library to support our specific use case while preserving the original behavior.

I am not very sure whether setting the zone to 'index' regardless of other factors could have side effects or not.

Reply:
There is a point to be made that the rest of the library is consistently using the interface name as the zone, so you're right that may not be the best approach.
But the current solution is a bit too hacky IMO.